### PR TITLE
Backend/profile page

### LINKED ISSUE #396: [Backend] Profile Page
### Issue

I will adjust userProfile endpoint to have a compact profile editing mechanism and have profile system detailed.

### Task(s)

- [x] Add public-private profile checks for getProfile.
- [x] Add check for whether the requester and the profile owner matches, for getProfile
- [x] Add is_private edit to updateProfile
- [x] Add name-surname edit to updateProfile
- [x] Add upload profile image endpoint
- [x] Add enrollments and created_spaces lists to getProfile
- [x] Add followUser endpoint
  - [x] Add followers and followed users to getProfile.

### Deliverable(s)

* PR for updates

### Acceptance Criteria

* PR is merged

### Deadline of the issue

30.11.2022 @23.59

### Reviewer

_No response_

### Deadline for Review

_No response_

### PR DESCRIPTION
**Pull request by ersoykadir**
Profile endpoint enlarged. Follow-unfollow added. Private-public profile added. Enrollments and createdSpaces lists added. Profile picture added. This PR closes #396

### CODE REVIEW

File: app/server/src/models/user/user.model.js [hunk 1]
```diff
@@ -105,12 +105,46 @@ const getPopulatedPersonalInfo = async (user_id) => {
...
+      select: { _id: 1, name: 1 }
+    })
+    .populate({
+      path: "follower_users",
+      select: { _id: 1, name: 1, surname: 1 }
+    })
+    .populate({
+      path: "followed_users",
+      select: { _id: 1, name: 1, surname: 1 }
+    })
```
Comment (furkanakkurt1335) [inline app/server/src/models/user/user.model.js:120]: This seems duplicate.

File: app/server/src/routes/userProfile/userProfile.controller.js [hunk 2]
```diff
@@ -37,23 +48,104 @@ const UserProfileController = {
...
+        if(user.followed_users.includes(id)){// looking at the profile of a user we follow
+          profile = await UserModel.getPopulatedPersonalInfo(id);
+          return res.status(200).json({ profile });
+        }
+        // looking at a profile we don't follow
+        const other_user = await UserModel.getUserByID(id);
+        let profile;
+        if(other_user.is_private){
+          profile = await UserModel.getPopulatedPersonalInfoPrivate(id);
+        }
```
Comment (furkanakkurt1335) [inline app/server/src/routes/userProfile/userProfile.controller.js:130]: Should we show the email and the picture of a private profile?

**Review by furkanakkurt1335 — CHANGES_REQUESTED** [review]
Small things before approval.

File: app/server/src/routes/userProfile/userProfile.controller.js (hunk 2 above)
Comment (ersoykadir) [reply app/server/src/routes/userProfile/userProfile.controller.js:130]: Yes, small image will be shown, email may be removed.
image

**Pull request merged**